### PR TITLE
GitHub Issue NOAA-EMC/GSI#408. Update default executable names

### LINKED ISSUE #408: Incorrect default executables
The executable names for `global_gsi.x`, `global_enkf.x`, and `ncdiag_cat.x` were all updated to new names, but the default executable names in global scripts weren't updated. These should be updated to reflect the new executable names.

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
The executable names for some executables have changed, but the default names in global scripts had not. These are now updated to the new names.

Fixes #408

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
Note: I haven't been able to test this because of the current issues with cycling in global workflow, but these are very straightforward updates.

File: scripts/exglobal_diag.sh [hunk 1]
```diff
@@ -47,7 +47,7 @@ export NLN=${NLN:-"/bin/ln -sf"}
 export CHGRP_CMD=${CHGRP_CMD:-"chgrp ${group_name:-rstprod}"}
 export NEMSIOGET=${NEMSIOGET:-${NWPROD}/exec/nemsio_get}
 export NCLEN=${NCLEN:-$HOMEgfs/ush/getncdimlen}
-export CATEXEC=${CATEXEC:-$HOMEgfs/exec/ncdiag_cat.x}
+export CATEXEC=${CATEXEC:-$HOMEgfs/exec/nc_diag_cat.x}
```
Comment (MichaelLueken) [inline scripts/exglobal_diag.sh]: This is not the correct name for ncdiag_cat.x.  The serial ncdiag_cat.x should be ncdiag_cat_serial.x.

**Review by MichaelLueken — COMMENTED** [review]
@WalterKolczynski-NOAA: These changes were made to `release/gfsda.v16.3.0` at [b19916b](https://github.com/NOAA-EMC/GSI/commit/b19916b6a9d180c6ecd1d318a116c625bf36ca5c).  The plan is to get these changes into the `develop` branch through the `gfsda.v16.3.0` release branch.

If these changes are required now, then please note that the correct executable for `scripts/exglobal_diag.sh` is `ncdiag_cat_serial.x` and not `nc_diag_cat.x`.

The GSI requires a single commit, so after changing the executable name, please use:

1) git add scripts/exglobal_diag.sh
2) git commit --amend
3) Save and close the commit message window
4) git push origin feature/update_exec_names --force

If you have any questions, please let me know.

File: scripts/exglobal_diag.sh (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply scripts/exglobal_diag.sh]: Okay, when I build I also get a `nc_diag_cat.x`, so I thought that was the correct one. I will fix.

**Comment by WalterKolczynski-NOAA** [comment]
It's not urgent, in the sense that cycling doesn't currently work right now anyway because of #348, but this change was just to make develop self-consistent. The executable names have already changed, but the scripts have not.

File: scripts/exglobal_diag.sh (hunk 1 above)
Comment (MichaelLueken) [reply scripts/exglobal_diag.sh]: Thanks for addressing this.  The GSI creates a parallel and serial executable for `ncdiag_cat`.  Before the update, `ncdiag_cat.x` was for the serial executable, while `ncdiag_cat_mpi.x` was the parallel version.  With the update, the parallel version became `nc_diag_cat.x` and the serial is `ncdiag_cat_serial.x`.

**Comment by MichaelLueken** [comment]
Since there are no changes to the source code, I will now give final approval to these changes and merge them to the authoritative `develop` branch.

**Pull request merged**